scalyr_agent/test_base.py

```diff
@@ -360,7 +360,11 @@ def tearDown(self):
             )
 
         if not self.__assertion_failed:
-            shutil.rmtree(self.logs_directory)
+            try:
+                shutil.rmtree(self.logs_directory)
+            except PermissionError:
+                # Workaround for Windows when sometimes deleting fails because file is still in use
```

Comment:
Could it be better to keep it failing and to try to close all opened descriptors?

Reply:
Yeah, that's why it's failing.

Problem is that it's not easy to keep track of all the open FDs to random files by various tests and ensure they are closed.

I will look again if I can identify tests which leave open FDs, but I may not be able to track down the root cause.

On Linux we could also use something like lsof or similar to find all the open FDs for this process and trying to close it, but that won't work on Windows...

Reply:
Looks like this did the trick - c3bc98a26f5e6eb54a710a5ecada8b20647b6ab0.

Main offender was ``scalyr_logging.set_log_destination`` which opens handles to the log files, but we (tests) never closed them.

I will also quickly audit other base test classes for this issue and fix them, where / if necessary.